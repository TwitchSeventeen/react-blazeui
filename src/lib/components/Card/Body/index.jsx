import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Body(props) {
  const {
    children, className, dataName, ...other
  } = props;
  const classes = classNames('c-card__body', className);
  return (
    <div
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </div>
  );
}

Body.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Body.defaultProps = {
  children: null,
  className: null,
  dataName: 'card-body',
};

export default Body;
