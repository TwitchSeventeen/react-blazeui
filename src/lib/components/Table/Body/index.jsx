import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Body(props) {
  const {
    children, className, dataName, as: Component, ...other
  } = props;

  const classes = classNames(
    'c-table__body',
    className,
  );

  return (
    <Component
      className={classes}
      {...other}
      data-name={dataName}
    >
      { children }
    </Component>
  );
}

Body.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
  /**
   * Element type to render as
   */
  as: PropTypes.elementType,
};

Body.defaultProps = {
  className: null,
  children: null,
  dataName: 'table-body',
  as: 'tbody',
};

export default Body;
