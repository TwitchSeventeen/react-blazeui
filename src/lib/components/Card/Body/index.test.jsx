import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Body from './index';

describe('Card.Body', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Body).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Body>
        <p>lorem</p>
      </Body>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
