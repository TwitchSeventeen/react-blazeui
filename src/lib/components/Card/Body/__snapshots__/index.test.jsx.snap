// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.Body renders correctly 1`] = `
<DocumentFragment>
  <div
    class="c-card__body"
    data-name="card-body"
  >
    <p>
      lorem
    </p>
  </div>
</DocumentFragment>
`;
